### ISSUE TITLE
Turtles get stuck in Halt mode

### ISSUE THREAD

**Issue by gamecocks08** [merlinlikethewizard/Mastermine#3]
So, someone before me had a similar problem but mine is that the turtles are stuck in halt mode and even when I tell them to do anything, they always return to halt. When they start, they move one block and then move back. When I turn on the power, nothing happens. The hub is constantly sending go_to_home directives. any ideas?

**Comment by merlinlikethewizard** [comment]
Sounds to me like they don't know their way home. So either they're too far away from the mine entrance (further than 8 blocks), or they're somehow below the mine entrance (entrance y value should be the same as the turtle on the ground). If neither of those things are true then lemme know and I'll think some more.

**Comment by gamecocks08** [comment]
yeah, the y value is the same as the turtle and they aren't far at all from the mine entrance, if you need screenshots let me know.

**Comment by merlinlikethewizard** [comment]
Dang, okay. Yeah, screenshots would be great (also cuz I'll get to see someone else having set it up). If you could send some of the layout, of the map screen, of the hub "whosmine" tab, and of the turtle "mastermine" tab, hopefully I'll be able to see something. I don't know if you're on a server or not but I'd also be happy to come on and take a gander. I'm really curious whats up, it sounds like it should be working...

**Comment by gamecocks08** [comment]
Yeah, I'm hosting a LAN world for some of my friends with logmein hamachi and using curseforge for the mods. If you have discord then I can send you the mods. I also am not home atm so I will get you the screenshots soon.

**Comment by gamecocks08** [comment]
Here are the [Screenshots](https://drive.google.com/drive/folders/1y1X7ca9f_6l2-5DBOLzgmBZRFOw3YJk4?usp=sharing).

**Comment by merlinlikethewizard** [comment]
Thanks! Sad to say though I'm stumped. Next thing to do would be to properly debug with print statements etc.

If you want a minecraft home visit my discord is MerlinLikeTheWizard#9300, I'd love to take a look :)

**Comment by SublimePhotographyuk** [comment]
Hey @merlinlikethewizard Im having the exact same issue, No matter what I do I can't seem to get it working unfortunately :/

**Comment by TheUnknown4004** [comment]
i have the same problem but mine doesnt say go_to_home directives the y value is correct its 6 blocks away from the mine entrance
it calibrated to 84,65,-255 but its supposed to be 83,65,-259 how can i fix this?

**Comment by merlinlikethewizard** [comment]
@TheUnknown4004 Well if it's calibrated wrong then probably the issue is with the gps setup. If even one coordinate is wrong on the gps host then all the gps requests will be off. You can check quickly on any wireless computer by using "gps locate". Lemme know if that's the issue or not. Come to think of it, @gamecocks08 you should probably check the same thing.

**Comment by TheUnknown4004** [comment]
i have check every coordinate there all correct i even did the gps host again i have dubble check the coordinates and it still calibrated to 84,65,-255

**Comment by gamecocks08** [comment]
@TheUnknown4004 just going to throw this out there, try moving the computer to a different position and try again, see if it's still off.

**Comment by DJ-666** [comment]
Did you make sure to edit the floppy disk size? That will cause the turtles to get just enough to go to home but not start mining.

Easy way to check and how I discovered it was by running "update" from the Terminal computer

**Comment by PanGeo-Code** [comment]
i found that if the turtles are stuck in halt a update command on the turtle fixes it at least update on the turtle worked for me

**Comment by http403** [comment]
I'm on 1.16 and have similar problem. The turtle will just stuck next to the disk drive after the calibration, except driving it manually. I can confirm the GPS works perfectly. I tried two different orientations and none work (one with hub facing north result in the turtle calibrate in the opposite direction).

**Comment by http403** [comment]
I'm on 1.16 and have similar problem. The turtle will just stuck next to the disk drive after the calibration, except driving it manually. I can confirm the GPS works perfectly with the pocket computer but the calibration always report one block ahead. I tried two different orientations and none work (north and east facing, one with hub facing north result in the turtle calibrate in the opposite direction). I tried manually move them away the disk drive and commend it go home but it just stuck in the same spot.

**Comment by PanGeo-Code** [comment]
> I'm on 1.16 and have similar problem. The turtle will just stuck next to the disk drive after the calibration, except driving it manually. I can confirm the GPS works perfectly with the pocket computer but the calibration always report one block ahead. I tried two different orientations and none work (north and east facing, one with hub facing north result in the turtle calibrate in the opposite direction). I tried manually move them away the disk drive and commend it go home but it just stuck in the same spot.
![Untitled](https://user-images.githubusercontent.com/81773780/125190730-da1b3d00-e23e-11eb-87bc-798973ed1b83.png)

see picture:
Grey is the monitor
Red is the advanced computer or the controll unit
yellow is how you place the turtle to set it up

and green is how you could tory to fix your halt problem

i had this too my monitor and computer where 1 block outside of the default 8x8 area that the turtles see as a appowed area so it would not move

**Comment by http403** [comment]
@PanGeo-Code Thanks for the quick reply. Your graph is really useful to dial in the setup. Now the turtle indeed moved around rather got stuck, but new problem appears. Now, turtles dive a very far away from the designated mining area (30 block+) and looping around the "prepare" directive (chunky) or "pass" (miner). I was in a flat snow biome, don't know if that affect the operations.

**Comment by PanGeo-Code** [comment]
> @PanGeo-Code Thanks for the quick reply. Your graph is really useful to dial in the setup. Now the turtle indeed moved around rather got stuck, but new problem appears. Now, turtles dive a very far away from the designated mining area (30 block+) and looping around the "prepare" directive (chunky) or "pass" (miner). I was in a flat snow biome, don't know if that affect the operations.

Then i do not Know, Maybe @merlinlikethewizard Know what your problem here is.

**Comment by http403** [comment]
I found the problem. It is due to the GPS setup. It wants one block above than what reported by the F3 debug window.
